contracts/javascore/lib/src/main/java/icon/proto/core/commitment/CommitmentProof.java

```diff
@@ -5,13 +5,13 @@
 import ibc.icon.score.util.ProtoMessage;
 
 public class CommitmentProof extends ProtoMessage {
-  private ExistenceProof exist;
+  private ExistenceProof exist = new ExistenceProof();
 
-  private NonExistenceProof nonexist;
+  private NonExistenceProof nonexist = new NonExistenceProof();
 
-  private BatchProof batch;
+  private BatchProof batch = new BatchProof();
 
-  private CompressedBatchProof compressed;
+  private CompressedBatchProof compressed = new CompressedBatchProof();
 
   public ExistenceProof getExist() {
```

Comment:
wont getExist always return proof if we initialize this by default?  Is this why decoding was failing?

Reply:
it would all be empty and we check for empty conditions. This is proto generated file btw.
The reason for failure in decoding was due to incorrect way of handling the var int array in proto decoder.